src/Common/Misc/ConvertProxyProperty/index.js

```diff
@@ -92,14 +92,15 @@ export function proxyPropToProp(property, ui) {
   const depStatus = depList ? Boolean(Number(depList.pop())) : true;
   const depValue = depList ? depList.pop() : null;
   const depId = depList ? depList.join(':') : null;
-  const searchString = [ui.name].concat(property.value).join(' ').toLowerCase();
+  const searchString = [ui.name].concat(property.value, depList).join(' ').toLowerCase();
 
   const prop = {
     show(ctx) {
+      let depTest = true;
       if (depId && ctx.properties[depId] !== undefined) {
-        return (ctx.properties[depId][0] === depValue) ? depStatus : !depStatus;
```

Comment:
Returning here may show properties just because they pass the "dependency" test, even though they would fail the "advanced" test.